### PR TITLE
refactor(react-grid): get rid of explicit dragging enabling

### PR DESCRIPTION
**Pull request by kvet**
BREAKING CHANGE:

In order to simplify API, we've decided to remove allowDragging properties in TableHeaderRow and GroupingPanel plugins. Now, to enable dragging it is enough to add the DragDropContext plugin.

### CODE REVIEW

File: packages/dx-react-grid/src/plugins/table-header-row.jsx [hunk 1]
```diff
@@ -46,27 +46,26 @@ export class TableHeaderRow extends React.PureComponent {
...
-                sorting, grouping, columns,
+                sorting, tableColumns, draggingEnabled,
               }, {
                 setColumnSorting, groupByColumn,
                 changeTableColumnWidths, changeDraftTableColumnWidths,
               }) => {
                 const { name: columnName } = params.tableColumn.column;
-                const groupingSupported = grouping !== undefined &&
-                    grouping.length < columns.length - 1;
+                const isNotTheLastVisible = tableColumns
```
Comment (SergeyAlexeev) [inline packages/dx-react-grid/src/plugins/table-header-row.jsx]: isTheLastInvisible?

**Comment by SergeyAlexeev** [comment]
Should we add BC? I mean the case when a user turns off dragging:

```
<TableHeaderRow allowDragging={false} />
```

And doesn't remove `<DragDropContext />`

Should we say: 'To disable table column dragging remove the DragDropContext plugin' ?

**Comment by kvet** [comment]
I've updated the PR description.

File: packages/dx-react-grid/src/plugins/table-header-row.jsx (hunk 1 above)
Comment (kvet) [reply packages/dx-react-grid/src/plugins/table-header-row.jsx]: Not sure that I understand how to use the find method here.

Comment (SergeyAlexeev) [reply packages/dx-react-grid/src/plugins/table-header-row.jsx]:
```js
const atLeastOneVisible = tableColumns.findIndex(({ type }) => type === TABLE_DATA_TYPE) !== -1;
```

Comment (kvet) [reply packages/dx-react-grid/src/plugins/table-header-row.jsx]: Ok, but I need to check that there are 2 and more columns.

Comment (SergeyAlexeev) [reply packages/dx-react-grid/src/plugins/table-header-row.jsx]: Oh, sorry, missed this `length > 1`. Anyway, `isNotTheLastVisible` sounds strange)

Comment (kvet) [reply packages/dx-react-grid/src/plugins/table-header-row.jsx]: I've changed it to `atLeastOneDataColumn`

**Pull request merged**